common/v2/features/SwapAssets/stateFactory.tsx

```diff
@@ -130,20 +133,27 @@ const SwapFlowFactory: TUseStateReducerFactory<SwapState> = ({ state, setState }
         isCalculatingFromAmount: true
       }));
 
-      const price = Number(
-        await DexService.instance.getTokenPriceTo(fromAsset.symbol, toAsset.symbol, value)
+      const commissionIncreasedAmount = withCommission({
+        amount: convert(Number(value)),
+        rate: MYC_DEXAG_COMMISSION_RATE
+      });
+
+      const { price, costBasis } = await DexService.instance.getTokenPriceTo(
+        fromAsset.symbol,
+        toAsset.symbol,
+        commissionIncreasedAmount.toString()
       );
 
       setState((prevState: SwapState) => ({
         ...prevState,
         isCalculatingFromAmount: false,
-        fromAmount: withCommission({
-          amount: convert(Number(value), price),
-          rate: MYC_DEXAG_COMMISSION_RATE
-        }).toString(),
+        fromAmount: (commissionIncreasedAmount * price).toString(),
```

Comment:
Quote from @vilsbole :
> Danger! Since we're dealing with 18 decimal values, and price is determined by an external provider this operation is unsafe and lead to an (over|under)flow. All multiplications and divisions MUST (RFC2119 be done using safe operations. In this project we use ethers.js [modified](https://ethereum.stackexchange.com/a/77898/24916) BN.js

Please use `convert` function for those operations